src/persistentstorage.cpp

```diff
@@ -258,3 +266,229 @@ void MysqlPersistentStorage::writelog(const std::string &id, const ObjectValue &
+                    logger_m.errorStream() << "couldn't create logging database: " << resp << endlog;
+            if (!_existsInArray(databases, persist_db_m))
+                if (!createDB(persist_db_m, resp))
+                    logger_m.errorStream() << "couldn't create persistence database: " << resp << endlog;
+        }
+    }
+}
+
+InfluxdbPersistentStorage::~InfluxdbPersistentStorage()
+{
+}
+
+static size_t _curlWriteCB(void *contents, size_t size, size_t nmemb, void *userp)
+{
+    ((std::string*)userp)->append((char*)contents, size * nmemb);
+    return size * nmemb;
+}
+
+#define INFLUXDB_EXTRA_DEBUG 0
+
+bool InfluxdbPersistentStorage::curlRequest(InfluxdbOperation_t oper, const std::string& db, const std::string& query, std::string& result)
+{
+    bool ret = false;
+    CURL *curl;
+	CURLcode res;
+	curl = curl_easy_init();
+	if(curl)
+	{
+		std::stringstream url;
+        url << uri_m << "/" << (oper == INFLUXDB_WRITE ? "write" : "query") << "?db=" << db;
+        if (!user_m.empty() && !pass_m.empty())
+        {
+            url << "&u=" << user_m << "&p=" << pass_m;
+        }
+        const char *field_data = query.c_str();
+
+#if INFLUXDB_EXTRA_DEBUG
+        logger_m.debugStream() << "requesting from " << url.str() <<  ": " << query << endlog;
+#endif
+
+		curl_easy_setopt(curl, CURLOPT_URL, url.str().c_str());
+        curl_easy_setopt(curl, CURLOPT_POSTFIELDS, field_data);
+        curl_easy_setopt(curl, CURLOPT_USERAGENT, "LinKNX");
+        curl_easy_setopt(curl, CURLOPT_WRITEFUNCTION, _curlWriteCB);
+        curl_easy_setopt(curl, CURLOPT_WRITEDATA, &result);
+
+		res = curl_easy_perform(curl);
+		if (res == CURLE_OK)
+        {
+            long response_code;
+            curl_easy_getinfo(curl, CURLINFO_RESPONSE_CODE, &response_code);
+            if (response_code / 100 == 2)
+                ret = true;
+            else
+                logger_m.errorStream() << "request to " << uri_m <<  " returned HTTP " << response_code << endlog;
+        }
+        else
+        {
+			logger_m.errorStream() << "request to " << uri_m <<  " failed: " << curl_easy_strerror(res) << endlog;
+        }
+#if INFLUXDB_EXTRA_DEBUG
+        logger_m.debugStream() << "response from db is: " << result << endlog;
+#endif
+		curl_easy_cleanup(curl);
+	}
+    return ret;
+}
+
+void InfluxdbPersistentStorage::exportXml(ticpp::Element* pConfig)
+{
+    pConfig->SetAttribute("type", "influxdb");
+    pConfig->SetAttribute("uri", uri_m);
+    pConfig->SetAttribute("user", user_m);
+    pConfig->SetAttribute("pass", pass_m);
+    pConfig->SetAttribute("db", db_m);
+    pConfig->SetAttribute("persist_db", persist_db_m);
+}
+
+void InfluxdbPersistentStorage::write(const std::string& id, const std::string& value)
+{
+    logger_m.infoStream() << "Persisting '" << value << "' for object '" << id << "'" << endlog;
+
+    std::stringstream query_s(std::ios_base::out|std::ios_base::binary);;
+    std::string resp;
+
+    if (value.find(':'))
```

Comment:
@fraxinas According to the [C++ reference](https://cplusplus.com/reference/string/string/find/), `find` returns [string::npos](https://cplusplus.com/string::npos) if no occurrence was found. I suggest to rewrite the condition to `value.find(':') != std::string::npos`.